core/src/main/kotlin/live/LiveGuild.kt

```diff
@@ -19,6 +20,124 @@ import dev.kord.core.event.user.VoiceStateUpdateEvent
 @KordPreview
 fun Guild.live(): LiveGuild = LiveGuild(this)
 
+@KordPreview
+inline fun Guild.live(block: LiveGuild.() -> Unit) = this.live().apply(block)
+
+@KordPreview
+fun LiveGuild.onEmojisUpdate(block: suspend (EmojisUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onIntegrationsUpdate(block: suspend (IntegrationsUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onBanRemove(block: suspend (BanRemoveEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onPresenceUpdate(block: suspend (PresenceUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onVoiceServerUpdate(block: suspend (VoiceServerUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onVoiceStateUpdate(block: suspend (VoiceStateUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onWebhookUpdate(block: suspend (WebhookUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onRoleCreate(block: suspend (RoleCreateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onRoleUpdate(block: suspend (RoleUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onRoleDelete(block: suspend (RoleDeleteEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMemberJoin(block: suspend (MemberJoinEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMemberUpdate(block: suspend (MemberUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMemberLeave(block: suspend (MemberLeaveEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+inline fun LiveGuild.onReaction(crossinline block: suspend (Event) -> Unit) = on<Event> {
```

Comment:
The methods who call directly "on" method cannot be inline, because in the "on" block is noinline.
So if you try to make the method inline, you need to make block crossinline, but with that, unable to compile because in "on" is noinline

Reply:
I understood that, but I don't understand why `onReaction` is inlined at all. 
`on` does not inline the given lambda, so even though this function is marked as inline the supplied lambda will still end up generating a SAM class. There's no generic to reify, there's no lambda to inline, what's the benefit?

Reply:
The inline keyword is apply for the block, because there is a SAM class created for the event, but the use of inline allows not to create a second SAM class for the block parameter.  This is why the crossinline keyword is used.
After, if you have a link to a documentation specifying, indeed, that this will create a second SAM class, I would like to have the link in order to perhaps better understand.

Reply:
We're on the same page here, both inline and non-inline versions result in the lambda not being inlined because of the non-inline-ness(?) of the `on` function. Each new invocation in the source code will result in a new class being created for both approaches.

If you wish to argue that inline the function results in one less object creation then I guess that's valid. It didn't came to mind at the time of asking but you're right.